### ISSUE TITLE
StartJobCommand on Amplify previews build with main branch HEAD instead of preview branch HEAD

### ISSUE THREAD

**Issue by dkandlyk** [aws/aws-sdk#653]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

When calling StartJobCommand from a preview branch and with a specific branch the rebuild pulls the main branch HEAD instead of preview branch latest commit.
Example of our data (the branchName is pulled from the ListBranchesCommand): 
```
{
    "appId": "***",
    "branchName": "pr-844",
    "jobType": "RELEASE"
}
```
The preview now has the changes from the Main branch instead of the specified branch

### SDK version number

@aws-sdk/client-amplify @version, 3.445.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

Node version 18.12.0

### Reproduction Steps

1. Create an Amplify app and add previews.
2. In a branch that is different from the main branch run 
```js
const amplifyClient = new AmplifyClient({ region: AWS_REGION });
  const listBranchesParams = {
    appId: appId,
    maxResults: 50,
  };
const branchList = await amplifyClient.send(new ListBranchesCommand(listBranchesParams));
Select any of the branches (not main)

const chosenBranch = responseAmplify.branches[1].branchName

const startJobParams: StartJobCommandInput = {
      appId: appId,
      branchName: chosenBranch,
      jobType: JobType.RELEASE,
    };
await amplifyClient.send(new StartJobCommand(startJobParams));
```
3. In the Amplify console go to --> previews --> pick the preview branch which changed --> build history --> see the build is from main branch HEAD instead of preview branch latest commit
It can also be seen from the response await amplifyClient.send(new StartJobCommand(startJobParams));
example response:
```
{
  "$metadata": {
    "httpStatusCode": 200,
    "requestId": "***",
    "cfId": "***",
    "attempts": 1,
    "totalRetryDelay": 0
  },
  "jobSummary": {
    "commitId": "HEAD",
    "commitTime": "2023-11-09T09:35:13.081Z",
    "jobArn": "***/pr-844/jobs/0000000003",
    "jobId": "3",
    "status": "PENDING"
  }
}
```

### Observed Behavior

When calling StartJobCommand through a preview branch with the JobType.Release. The preview branch build pulls the head of the main branch instead of the preview branch head

### Expected Behavior

It should build with the preview branch head

**Comment by RanVaknin** [comment]
Hi @dkandlyk ,


Understood. Here's a more direct and technical response to the customer, based on the information from the Amplify team:

Hello,

I've received feedback from the Amplify team regarding the issue you're experiencing with the StartJob operation in conjunction with PR previews. Here's the key takeaway:

The Amplify team has acknowledged a bug where using the `StartJob` command along with enabled PR previews leads to builds being incorrectly based off the target branch. However, they also noted that the simultaneous use of PR previews and manual `StartJob` hooks is not a recommended practice.

In typical scenarios, if PR previews are enabled and a PR is open, each new commit to that branch should automatically trigger its own deployment, making the manual invocation of StartJob unnecessary. The Amplify team is interested in understanding if there is a specific reason or unique use case behind your approach of using both features together.

If you have particular use case for this setup, please let us know. This information will help the Amplify team understand the context and explore possible solutions or workarounds for your situation.

Thanks,
Ran~

**Comment by dkandlyk** [comment]
Hi @RanVaknin! :)
Sorry, for the late response, I have been on Christmas holiday. :)
Our use case:
We have a mono repo with a Nextjs application. We have created a page where team members can switch `environment values` to support switching API's and secrets on preview branches. When they change `environment values` it triggers the  `StartJob` because Nextjs needs the `environment values` on build time.

So a scenario for us could be a team member who has made changes to a dev API and creates a new preview branch. To test it he switches  the `environment value` to look at the dev API instead of the prod which triggers the `StartJob` so Nextjs can do a rebuild with the dev API `environment value`.

So in these cases we do not make a commit we only need a rebuild of the application. :)

Hope it makes sense :)
-Anders

**Comment by RanVaknin** [comment]
I have not heard back from the API team, but I can see that this is on their backlog. 

Since this is not actionable by the SDK team, I'm going to go ahead and close this. 
I would keep an eye on your environment to see if this is being solved automatically (should be a service side fix, so no SDK update will be needed).

If you have access to support via the AWS console, feel free to refer to the ticket ID I created with that team for updates (`P107055715`).

Thanks again,
Ran~
